packages/notifi-frontend-client/lib/NotifiFrontendClient.ts

```diff
@@ -73,4 +81,61 @@ export class NotifiFrontendClient {
 
     await Promise.all([saveAuthorizationPromise, saveRolesPromise]);
   }
+
+  async beginLoginViaTransaction({
+    walletBlockchain,
+    walletAddress,
+  }: BeginLoginProps): Promise<Types.BeginLogInByTransactionResult> {
+    const { dappAddress } = this._configuration;
+
+    const result = await this._service.beginLogInByTransaction({
+      walletAddress: walletAddress,
+      walletBlockchain: walletBlockchain,
+      dappAddress,
+    });
+
+    const nonce = result.beginLogInByTransaction.nonce;
+
+    if (nonce !== null) {
+      const ruuid = crypto.randomUUID();
+      this._clientRandomUuid = ruuid;
+      const encoder = new TextEncoder();
+      const data = encoder.encode(nonce + ruuid);
+
+      const hashBuffer = await crypto.subtle.digest('SHA-256', data);
+
+      const hashArray = Array.from(new Uint8Array(hashBuffer));
+      const logValue =
+        'Notifi Auth: 0x' +
+        hashArray.map((b) => b.toString(16).padStart(2, '0')).join('');
+      return { nonce: logValue };
+    } else throw new Error('Failed to begin login process');
+  }
+
+  async completeLoginViaTransaction({
+    walletBlockchain,
+    walletAddress,
+    transactionSignature,
+  }: CompleteLoginProps): Promise<Types.CompleteLogInByTransactionMutation> {
+    const { dappAddress } = this._configuration;
+    const clientRandomUuid = this._clientRandomUuid;
```

Comment:
after this assignment, I think we should clear out `this._clientRandomUuid` -- what do you think?

Reply:
Yes, I think that clearing it out makes sense in case `beginViaTransaction` has to run again in the same SDK instance.

```
const clientRandomUuid = this._clientRandomUuid;

this._clientRandomUuid = null;
```